Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -856,90 +953,137 @@ namespace Terrain
             }
         }
 
-        float width = (request.m_samplesX - 1) * request.m_vertexSpacing;
-        float height = (request.m_samplesY - 1) * request.m_vertexSpacing;
-        AZ::Vector3 aabbMin = AZ::Vector3(request.m_worldStartPosition.GetX(), request.m_worldStartPosition.GetY(), m_worldBounds.GetMin().GetZ() + minHeight);
-        AZ::Vector3 aabbMax = AZ::Vector3(aabbMin.GetX() + width, aabbMin.GetY() + height, m_worldBounds.GetMin().GetZ() + maxHeight);
-        meshAabb.Set(aabbMin, aabbMax);
+        if (maxHeight < minHeight)
+        {
+            // All height samples were invalid, so set the aabb to null.
+            meshAabb.SetNull();
+        }
+        else
+        {
+            float width = (request.m_samplesX - 1) * request.m_vertexSpacing;
+            float height = (request.m_samplesY - 1) * request.m_vertexSpacing;
+            AZ::Vector3 aabbMin = AZ::Vector3(request.m_worldStartPosition.GetX(), request.m_worldStartPosition.GetY(), m_worldBounds.GetMin().GetZ() + minHeight);
+            AZ::Vector3 aabbMax = AZ::Vector3(aabbMin.GetX() + width, aabbMin.GetY() + height, m_worldBounds.GetMin().GetZ() + maxHeight);
+            meshAabb.Set(aabbMin, aabbMax);
+        }
     }
 
-    void TerrainMeshManager::ProcessSectorUpdates(AZStd::span<SectorUpdateContext> sectorUpdates)
+    void TerrainMeshManager::ProcessSectorUpdates(AZStd::vector<AZStd::vector<StackSectorData*>>& sectorUpdates)
     {
         AZ::JobCompletion jobCompletion;
-        for (SectorUpdateContext& updateContext : sectorUpdates)
+
+        for (uint32_t lodLevel = 0; lodLevel < sectorUpdates.size(); ++lodLevel)
         {
-            const float gridMeters = (GridSize * m_sampleSpacing) * (1 << updateContext.m_lodLevel);
-            StackSectorData* sector = updateContext.m_sector;
+            auto& sectors = sectorUpdates.at(lodLevel);
+            if (sectors.empty())
+            {
+                continue;
+            }
 
-            const auto jobLambda = [this, sector, gridMeters]() -> void
+            for (StackSectorData* sector : sectors)
             {
-                AZStd::vector<HeightNormalVertex> meshHeightsNormals;
+                const float gridMeters = (GridSize * m_sampleSpacing) * (1 << lodLevel);
 
+                const auto jobLambda = [this, sector, gridMeters]() -> void
```

Comment:
Suggestion: Consider pulling the lambda out into a separate method to "de-bulk" this one a little bit and make things easier to read.

Reply:
I think if I move the AABB stuff out to its own function it'll make the size of the job lambda less annoying.